### PR TITLE
Fix mesos stats instance number calculation, add new memory metrics

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

File: mesos_stats/mesos.py [hunk 1]
```diff
@@ -210,10 +213,10 @@ def new_slave_task_metrics(mesos, requests_json=None):
                             break
                     else:
                         # We should never get here but just in case
-                        task_instance = t['executor_id'].split('-mesos-slave', 1)[0]
+                        task_instance = t['executor_id'].split('-mesos_slave', 1)[0]
```
Comment (pennello) [inline mesos_stats/mesos.py]: I take it this was a bug.

File: mesos_stats/mesos.py [hunk 2]
```diff
@@ -151,6 +151,9 @@ def old_slave_task_metrics(mesos):
         Metric("cpus_limit",            prefix + ".cpus.limit"),
         Metric("mem_limit_bytes",       prefix + ".mem.limit_bytes"),
         Metric("mem_rss_bytes",         prefix + ".mem.rss_bytes"),
+        Metric("mem_anon_bytes",        prefix + ".mem.anon_bytes"),
+        Metric("mem_file_bytes",        prefix + ".mem.file_bytes"),
+        Metric("mem_mapped_file_bytes", prefix + ".mem.mapped_file_bytes"),
```
Comment (pennello) [inline mesos_stats/mesos.py]: What's with this function being for "old style" metrics?  Does this mean the metrics you added here should also be in the "new" arena, whatever that is?

File: mesos_stats/mesos.py (hunk 1 above)
Comment (mikebell90) [inline mesos_stats/mesos.py]: Recommend you document your ASSUMPTION as to format.

File: mesos_stats/mesos.py (hunk 2 above)
Comment (stevenschlansker) [reply mesos_stats/mesos.py]: whoops, good catch

Comment (mikebell90) [reply mesos_stats/mesos.py]: ah yes, good point

File: mesos_stats/mesos.py (hunk 1 above)
Comment (stevenschlansker) [reply mesos_stats/mesos.py]: Really, this should just be redone.  It's going to break again if we ever rename our machines to `agent` or ever have a task with 10 or more instances...

Comment (stevenschlansker) [reply mesos_stats/mesos.py]: Correct:
screen shot 2018-03-23 at 9 23 15 am

File: mesos_stats/mesos.py [hunk 3]
```diff
@@ -191,6 +191,9 @@ def new_slave_task_metrics(mesos, requests_json=None):
         Metric("cpus_limit", tc_prefix + ".cpus.limit"),
         Metric("mem_limit_bytes", tc_prefix + ".mem.limit_bytes"),
         Metric("mem_rss_bytes", tc_prefix + ".mem.rss_bytes"),
+        Metric("mem_anon_bytes",        tc_prefix + ".mem.anon_bytes"),
+        Metric("mem_file_bytes",        tc_prefix + ".mem.file_bytes"),
+        Metric("mem_mapped_file_bytes", tc_prefix + ".mem.mapped_file_bytes"),
```
Comment (pennello) [inline mesos_stats/mesos.py]: Maybe if you're going to line these up, consider lining up the pre-existing lines.

File: mesos_stats/mesos.py (hunk 1 above)
Comment (jay-chin) [reply mesos_stats/mesos.py]: This should be fixed in the latest mesos_stats change

**Comment by jay-chin** [comment]
It looks like those mem metrics have been deprecated since 0.24 ( https://gist.github.com/chetan/9c519c68549b55d71709cb8cc62206ae#cgroupsmem) . I've checked on one of our running slave instance and they are no longer there. Too bad the official documentation does not show this :-(

http://mesos-slave29-qa-sf.qasql.opentable.com:5051/monitor/statistics.json

**Comment by jay-chin** [comment]
Closing as those memory metrics have been deprecated

**Pull request closed without merging**